### PR TITLE
feat: add player kill notifier

### PR DESCRIPTION
**Pull request by iProdigy**
image

```json
{
  "type":"PLAYER_KILL",
  "playerName":"dankgg",
  "accountType":"NORMAL",
  "extra":{
    "name":"Shipy",
    "combatLevel":74,
    "equipment":{
      "AMULET":{
        "id":1731,
        "quantity":1,
        "priceEach":1987,
        "name":"Amulet of power"
      },
      "WEAPON":{
        "id":1333,
        "quantity":1,
        "priceEach":14971,
        "name":"Rune scimitar"
      },
      "TORSO":{
        "id":1135,
        "quantity":1,
        "priceEach":4343,
        "name":"Green d\u0027hide body"
      },
      "LEGS":{
        "id":1099,
        "quantity":1,
        "priceEach":2077,
        "name":"Green d\u0027hide chaps"
      },
      "HANDS":{
        "id":1065,
        "quantity":1,
        "priceEach":1392,
        "name":"Green d\u0027hide vambraces"
      }

    },
    "world":394,
    "location":{
      "x":3334,
      "y":4761,
      "plane":0
    },
    "myHitpoints":20,
    "myLastDamage":12
  }
}
```

### CODE REVIEW

**Comment by iProdigy** [comment]
Known shortcoming: if a player dies to poision (while we hit 0 on the same tick), the notifier will *not* be triggered because we can't (easily) distinguish whether that poison was inflicted by the local player or another source.

File: src/main/java/dinkplugin/notifiers/data/PkNotificationData.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@EqualsAndHashCode(callSuper = false)
+public class PkNotificationData extends NotificationData {
+
+    String name;
+    int combatLevel;
+    Map<KitType, SerializedItemStack> equipment;
+    int world;
+    WorldPoint location;
+    int myHitpoints;
+    int myLastDamage;
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/data/PkNotificationData.java:23]: Should any of these variables be available for the string template replacement (e.g., `%LEVEL%`)?

File: src/main/java/dinkplugin/notifiers/data/PkNotificationData.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.List;
+import java.util.Map;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class PkNotificationData extends NotificationData {
+
+    String name;
+    int combatLevel;
+    Map<KitType, SerializedItemStack> equipment;
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/data/PkNotificationData.java:19]: Should these be named differently to make it clearer that they refer to the victim's info?

File: README.md [hunk 3]
```diff
@@ -524,6 +525,62 @@ Note: `petName` is only included if the game sent it to your chat via untradeabl
 ```
 </details>
 
+### Player Kills:
+
+`%TARGET%` will be replaced with the victim's user name
```
Comment (pajlada) [inline README.md:530]: Include note that `world` and `location` will be omitted if the toggle is enabled/disabled
You can also make it an entirely new example - up to but I think both options are clear

File: src/main/java/dinkplugin/notifiers/data/PkNotificationData.java (hunk 1 above)
Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/data/PkNotificationData.java:23]: I think we can skip that until a request for it comes through

File: src/main/java/dinkplugin/notifiers/data/PkNotificationData.java (hunk 2 above)
Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/data/PkNotificationData.java:19]: Yeah, `victimEquipment` seems reasonable. Remember to update the readme at the same time

File: src/main/java/dinkplugin/notifiers/data/PkNotificationData.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package dinkplugin.notifiers.data;
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/data/PkNotificationData.java:1]: Class & file name are `PkNotification` while the notifier is `PlayerKill..`  - can this be renamed for consistency?

**Review by pajlada — CHANGES_REQUESTED** [review]
Code looks good to me :+1:

**Review by pajlada — APPROVED** [review]
Did simple tests and things look good from my perspective :+1:

**Pull request merged**